New addon: Debuffed

aukon opened the pull request.

aukon committed:
Debuffed Addon

aukon committed:
Create README.md

aukon committed:
Update addons.xml

z16 commented on addons/Debuffed/Debuffed.lua:
```diff
@@ -0,0 +1,250 @@
+--[[
+Copyright (c) 2018, Auk
```
Please use the proper copyright sign `©`.

z16 commented on addons/Debuffed/Debuffed.lua:
```diff
@@ -0,0 +1,250 @@
+    * Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer in the
+    documentation and/or other materials provided with the distribution.
+    * Neither the name of <addon name> nor the
```
Add your addon name here.

z16 commented on addons/Debuffed/Debuffed.lua:
```diff
@@ -0,0 +1,250 @@
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL <your name> BE LIABLE FOR ANY
```
And your name again here.

z16 commented on addons/Debuffed/debuffs.lua:
```diff
@@ -0,0 +1,54 @@
+    return {
```
You know, I think this would be a useful addition to Windower resources. Feel free to add this to the [ResourceExtractor](https://github.com/Windower/ResourceExtractor/) repository (in the [fixes.xml](https://github.com/Windower/ResourceExtractor/blob/master/fixes.xml) file). That way you could access that then.

I would also leave out the name field (both if you go the ResourceExtractor route or if you leave it here). You can use the ID to access the name from the buff resources, and then you can use the `.name` property to localize it based on client language (so it shows the japanese name on the japanese client). You can still use the english name in a comment at the end of the line for easier maintenance (both here and in the `fixes.xml` file).

z16 commented on addons/Debuffed/Debuffed.lua:
```diff
@@ -0,0 +1,250 @@
+            end
+        end
+    end
+    box.current_string = count > 0 and current_string or ''
```
This is minor suggestion, but repeated expanding string concatenation can be "very" expensive (for certain definitions of "very"). I would probably use a list to append lines to and later join them all by `\n`.

Has the even minorer additional benefit, that you won't need the count variable - at the end of the function you could just check if the table is empty. If so - return. If not, update the `box.current_string` with `Debuffed [' .. target.name ..']\n' .. list:concat('\n')`, where `lines` is the table used to hold all the lines.

That makes it a tad more efficient, but I doubt it would make much of a difference unless it's on the lowest of low boxes.

z16 commented on addons/Debuffed/Debuffed.lua:
```diff
@@ -0,0 +1,250 @@
+end
+
+function handle_overwrites(target, new, t)
+    if debuffed_mobs[target] then
```
Can invert the condition to reduce nesting:
```lua
    if not debuffed_mobs[target] then
        return true
    end
```

z16 commented on addons/Debuffed/Debuffed.lua:
```diff
@@ -0,0 +1,250 @@
+end
+
+function inc_action(act)
+    if act.category == 4 then
```
Same as above, can invert the condition:
```lua
    if act.category ~= 4 then
        return
    end
```

z16 commented on addons/Debuffed/Debuffed.lua:
```diff
@@ -0,0 +1,250 @@
+        print('    \\cs(255,255,255)interval <value>\\cr - Allows you to change the refresh interval (default: 0.1)')
+        print('    \\cs(255,255,255)blacklist|whitelist add|remove <name>\\cr - Adds or removes the spell <name> to the specified list')
+    end
+end)
```
No new line at the end of the file. This is critical.

z16 reviewed (changes requested):
Overall it looks pretty good, you just need to fix the copyright notice. Aside from that all the things I commented on are minor suggestions, although I would like to see the debuff file committed to the RE repository. But even if you decide to do that, it wouldn't have to be now, can do it at any other point in time.

So once you fix the copyright notice let me know if you want me to merge it as is, or if you want to implement any of the other suggestions.

aukon committed:
Code cleanup

aukon committed:
Add timers command

aukon commented:
Implemented all of those suggestions. I'll definitely look into the resources but I still need to add a few other spells like songs yet.

aukon committed:
Hide timers below 0

aukon committed:
Revert "Hide timers below 0"

This reverts commit b0c813dea003ba61f5a922aebc46c064e08cde51.

aukon committed:
Hide timers below 0

aukon committed:
Use resources for debuffs

aukon commented:
Switched to using resources now and I've made a pull request with an updated fixes.xml. https://github.com/Windower/ResourceExtractor/pull/52/commits/39b324adcef04f5acd7c5e928d32eee2b77afaaf

aukon committed:
Fix debuffs not being cleared

z16 commented:
Just so you know, I want to refactor RE first to handle tables within the fixes file and interpret it as an array. As it is now we end up with the string `'{123,456,789}'` instead of an actual table in Lua. After that you'll need to adjust your code as well at the appropriate place. Since I'm currently on vacation this is taking a bit, sorry for the delay...

aukon commented:
No worries, I did take a look at RE myself but it's a bit out of my expertise. Should be a lot cleaner than the hacky workaround I used for the string issue.

aukon committed:
Merge branch 'dev' into dev

z16 commented:
Sorry this took so long, we went over a few things and resources were adjusted again. Now the overwrites should be an actual table instead of a string containing a table definition.

Also adjusted some duration values, but not many. Some seem to be off still, but it should be usable. Let me know when you've upgraded the addon to handle the new resources.

aukon committed:
Update to handle new resources

aukon commented:
Fixed for the new resources now but I also added a check for Light Shot to upgrade Dia. I've made two more pull requests to add the spell attributes for Dia IV so overwrites work properly for it.

https://github.com/Windower/ResourceExtractor/pull/54
https://github.com/Windower/Resources/pull/28

I'll look into fixing the durations if I ever get chance, haven't been actively playing recently.
